test/Tester/HeterogeneousSilosTests/UpgradeTests/UpgradeTests.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Threading.Tasks;
+using Orleans.Versions.Compatibility;
+using Orleans.Versions.Placement;
+using TestVersionGrainInterfaces;
+using Xunit;
+
+namespace Tester.HeterogeneousSilosTests.UpgradeTests
+{
+    [TestCategory("Versioning"), TestCategory("ExcludeXAML"), TestCategory("SlowBVT"), TestCategory("Functional")]
+    public class MinimumVersionTests : UpgradeTestsBase
+    {
+        protected override VersionPlacementStrategy VersionPlacementStrategy => MinimumVersionPlacement.Singleton;
+        protected override VersionCompatibilityStrategy VersionCompatibilityStrategy => BackwardCompatible.Singleton;
+        
+        [Fact]
+        public Task AlwaysCreateActivationWithMinimumVersionTest()
+        {
+            return Step1_StartV1Silo_Step2_StartV2Silo_Step3_StopV2Silo(1);
+        }
+    }
+
+    [TestCategory("Versioning"), TestCategory("ExcludeXAML"), TestCategory("SlowBVT"), TestCategory("Functional")]
+    public class LatestVersionTests : UpgradeTestsBase
+    {
+        protected override VersionPlacementStrategy VersionPlacementStrategy => LatestVersionPlacement.Singleton;
+        protected override VersionCompatibilityStrategy VersionCompatibilityStrategy => BackwardCompatible.Singleton;
+
+        [Fact]
+        public Task AlwaysCreateActivationWithLatestVersionTest()
+        {
+            return Step1_StartV1Silo_Step2_StartV2Silo_Step3_StopV2Silo(2);
+        }
+
+        [Fact]
+        public Task UpgradeProxyCallNoPendingRequestTest()
```

Comment:
>UpgradeProxyCallNoPendingRequest [](start = 20, length = 32)

Can you explain a little bit more what you are asserting? especially since the assert is in external call